[Bug] Compilation steps for Linux fail with CMake errors

#### Describe the issue

When following the "How To Compile" instructions for Linux from the wiki, I get the errors below at the 
"[Building the program (configuring)](https://github.com/CorsixTH/CorsixTH/wiki/How-To-Compile#building-the-program-configuring)"-step.

Prior to running CMake, I'd installed the requirements via apt-get as per the instructions on the same wiki page.

I am not particularly familiar with CMake and cannot decipher what the problem may be, so I'd appreciate some help so we can maybe improve the experience for someone wanting to compile CorsixTH themselves.

#### Steps to Reproduce

```sh
$ git clone https://github.com/CorsixTH/CorsixTH
Cloning into 'CorsixTH'...
remote: Enumerating objects: 28611, done.
remote: Total 28611 (delta 0), reused 0 (delta 0), pack-reused 28611
Receiving objects: 100% (28611/28611), 23.89 MiB | 1.07 MiB/s, done.
Resolving deltas: 100% (23427/23427), done.

$ cd CorsixTH/ && ls
AnimView      changelog.txt  CMakeLists.txt    CorsixTH          DoxyGen       LDocGen    libs         README.md   scripts        WindowsInstaller
appveyor.yml  CMake          CONTRIBUTING.txt  Dockerfile.build  ground_tiles  LevelEdit  LICENSE.txt  README.txt  SpriteEncoder

$ mkdir build
$ cd build/
$ cmake ..
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Note: SDL audio is enabled (default)
Note: FFmpeg video is enabled (default)
Note: FreeType2 is enabled (default)

Building common libraries
You have called ADD_LIBRARY for library RncLib without any source files. This typically indicates a problem with your CMakeLists.txt file
Building CorsixTH
You have called ADD_LIBRARY for library CorsixTH_lib without any source files. This typically indicates a problem with your CMakeLists.txt file
CMake Error at CorsixTH/CMakeLists.txt:48 (add_executable):
  add_executable called with incorrect number of arguments


CMake Error at CorsixTH/CMakeLists.txt:52 (target_link_libraries):
  Cannot specify link libraries for target "CorsixTH" which is not built by
  this project.


-- Configuring incomplete, errors occurred!
See also "/home/user/games/CorsixTH/build/CMakeFiles/CMakeOutput.log".
```

#### Expected Behavior

I expect CMake to complete successfully.

#### System Information
**CorsixTH Version:** 8741d41e ( v0.64 )

**Operating System:** elementaryOS 5.1 - derived from Ubuntu 18.04 LTS

**Theme Hospital Version:** GOG.com

### Additional Info

`CMakeOutput.log` does not seem particularly interesting, but I've [uploaded it here anyway](https://gist.github.com/achton/7753b6a2ebc1336822b22f36c9a1cc32).

```sh
$ cmake --version
cmake version 3.10.2
```

```sh
$ g++ --version
g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/CorsixTH/CMakeLists.txt
+++ b/CorsixTH/CMakeLists.txt
@@ -26,8 +26,6 @@ else()
   set(CORSIX_TH_INTERPRETER_PATH ${CMAKE_INSTALL_FULL_DATADIR}/corsix-th/CorsixTH.lua)
 endif()
 
-add_library(CorsixTH_lib STATIC)
-
 # Declaration of the executable
 if(APPLE)
   set(corsixth_icon_file ${CMAKE_SOURCE_DIR}/CorsixTH/Icon.icns)
@@ -48,6 +46,13 @@ else()
   add_executable(CorsixTH)
 endif()
 
+# Ensure config.h is picked up by cmake - moving this into subdir cmake files will
+# prevent it applying to the CorsixTH project
+set(CorsixTH_generated_src_dir ${CMAKE_BINARY_DIR}/CorsixTH/Src/)
+add_library(CorsixTH_lib STATIC ${CorsixTH_generated_src_dir})
+
+target_include_directories(CorsixTH_lib PUBLIC ${CorsixTH_generated_src_dir})
+
 target_link_libraries(CorsixTH_lib RncLib)
 target_link_libraries(CorsixTH CorsixTH_lib)
 
@@ -59,10 +64,6 @@ endif()
 # Note: Done after generating targets
 add_subdirectory(${PROJECT_SOURCE_DIR}/Src)
 add_subdirectory(${PROJECT_SOURCE_DIR}/SrcUnshared)
-
-# Ensure config.h is picked up by cmake - moving this into subdir cmake files will
-# prevent it applying to the CorsixTH project
-target_include_directories(CorsixTH_lib PUBLIC ${CMAKE_BINARY_DIR}/CorsixTH/Src/)
 
 # Set language standard
 set_property(TARGET CorsixTH_lib PROPERTY CXX_STANDARD 14)

